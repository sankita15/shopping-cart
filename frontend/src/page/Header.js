import React from 'react';
import { Nav, NavLink, NavItem } from 'reactstrap';
import { FaViadeo } from 'react-icons/fa';
import { GoSignOut } from 'react-icons/go';

export default class Header extends React.Component {
    render() {
        return (
            <div className="header">
                <Nav>
                    <NavItem>
                        <NavLink className="icon">
                            <FaViadeo size={45} />
                        </NavLink>
                    </NavItem>
                    <NavItem>
                        <NavLink href="/products" className="link">
                            Home
                        </NavLink>
                    </NavItem>
                    <NavItem>
                        <NavLink href="/logout" className="log-out">
                            <GoSignOut size={25} />
                        </NavLink>
                    </NavItem>
                </Nav>
            </div>
        );
    }
}
