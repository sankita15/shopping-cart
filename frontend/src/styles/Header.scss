.header{
  background-color: #232f3e;
  color: white;
  font-weight: bold;
  height: 66px;
  padding-top: 12px;
  margin-bottom: 25px;
}
.link{
  cursor: pointer;
  margin-top: 10px;
  color: white;

  &:hover{
    border-radius: 5px;
    border: 0.5px solid lightgrey;
    color: white;
  }
}

.log-out{
  color: white;
  margin-top: 6px;
  margin-left: 1220px;

  &:hover{
    color: gray;
  }
}

.icon{
padding-top: 0px;
  padding-right: 15px;
}
