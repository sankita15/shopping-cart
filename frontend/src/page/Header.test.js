import React from 'react';
import { shallow } from 'enzyme';
import { Nav, NavLink, NavItem } from 'reactstrap';
import { FaViadeo } from 'react-icons/fa';
import { GoSignOut } from 'react-icons/go';
import Header from './Header';

describe('Header Description', () => {
    const createWrapper = () => shallow(<Header />);

    describe('render header', () => {
        it('should match snapshot', () => {
            const component = createWrapper();

            expect(component).toMatchSnapshot();
        });

        it('should render Navs for header', () => {
            const component = createWrapper();

            const icon = component.find(Nav).find(NavItem).at(0).find(NavLink)
                .find(FaViadeo);
            const home = component.find(Nav).find(NavItem).at(1).find(NavLink);
            const logout = component.find(Nav).find(NavItem).at(2).find(NavLink)
                .find(GoSignOut);

            expect(icon.length).toEqual(1);
            expect(home.length).toEqual(1);
            expect(home.prop('children')).toEqual('Home');
            expect(logout.length).toEqual(1);
        });
    });

    describe('Home Tab', () => {
        it('should redirect to product page when click on Home Tab', () => {
            const component = createWrapper();
            const home = component.find(Nav).find(NavItem).at(1).find(NavLink);
            const homeTabLink = home.prop('href');

            expect(homeTabLink).toBe('/products');
        });
    });

    describe('LogOut Icon', () => {
        it('should log out when click on Profile icon and select log out', () => {
            const component = createWrapper();
            const logout = component.find(Nav).find(NavItem).at(2).find(NavLink);
            const logOutRedirection = logout.prop('href');

            expect(logOutRedirection).toBe('/logout');
        });
    });
});
