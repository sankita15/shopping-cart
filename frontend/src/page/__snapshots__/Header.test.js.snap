// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Description render header should match snapshot 1`] = `
<div
  className="header"
>
  <Nav
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="icon"
        tag="a"
      >
        <FaViadeo
          size={45}
        />
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className="link"
        href="/products"
        tag="a"
      >
        Home
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className="log-out"
        href="/logout"
        tag="a"
      >
        <GoSignOut
          size={25}
        />
      </NavLink>
    </NavItem>
  </Nav>
</div>
`;
